unified-runtime/test/layers/tracing/hello_world_collected.match

```diff
@@ -0,0 +1,26 @@
+RUN: %use-mock %xptienable XPTI_SUBSCRIBERS=%{shlibpre}collector%{shlibext} hello_world 2>&1 | %filecheck %s
+REQUIRES: filecheck
```

Comment:
I really dislike this unnatural filecheck handling... Can we just assume it's available and spell as `FileCheck` without using substitutions?

Reply:
For some reason, the Python package spells it as `filecheck` while LLVM spells it as `FileCheck`. There's also the possibility that the user wants to specify a specific FileCheck library using `URLIT_FILECHECK_BINARY` (for example, if they have an LLVM build elsewhere).

I could add `FileCheck` as a substitution (i.e. without the `%`), but that would hide the fact that it's a substitution which could be confusing. It also could cause problems if someone creates a file named like `do_thing_filecheck.match` or similar.

Reply:
Ah, thanks for pointing that out; didn't realise that LLVM was using `FileCheck` as a substitution. I've implemented a reduced version of the logic and now `| FileCheck %s` can be used.